### ISSUE TITLE
az upgrade failed when az --version returning "You have 1 updates available. Consider updating your CLI installation with 'az upgrade'"

### ISSUE THREAD

**Issue by happyiphone** [Azure/azure-cli#15133]
## Because Azure CLI is a lot more trusted by us than anything else we automated our processes using Azure CLI, because of this we check the Azure CLI version installed before we do anything with it and if it is outdated we prevent the user from continuing, because of this we found out that after 2.11.1 the `az --version` is always coming up with a `You have 1 updates available. Consider updating your CLI installation with 'az upgrade'` even when we have the latest version installed.

When we try to run `az upgrade` we get back 

`You already have the latest azure-cli version: 2.11.1
Checking update for interactive
No updates available for 'interactive'. Use --debug for more information.
Upgrade finished.`

**Command Name**
`az --version`

**Errors:**```
azure-cli                         2.11.1
core                              2.11.1
**telemetry                          1.0.5 ***
Extensions:
interactive                        0.4.4

Python location 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\jherrera\.azure\cliextensions'

Python (Windows) 3.6.8 (tags/v3.6.8:3c6b436a57, Dec 23 2018, 23:31:17) [MSC v.1916 32 bit (Intel)]

Legal docs and information: aka.ms/AzureCliLegal

You have 1 updates available. Consider updating your CLI installation with 'az upgrade'

Please let us know how we are doing: https://aka.ms/azureclihats
and let us know if you're interested in trying out our newest features: https://aka.ms/CLIUXstudy

```
## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- ``

## Expected Behavior

## Environment Summary
```
Windows-10-10.0.18362-SP0
Python 3.6.8
Installer: MSI

azure-cli 2.11.1

Extensions:
interactive 0.4.4

```
## Additional Context

<!--Please don't remove this:-->
<!--auto-generated-->

**Comment by yungezz** [comment]
thanks for reporting. hi @fengzhou-msft could you pls have a look?

**Comment by DAD713** [comment]
FYI: Having the same issue

**Comment by fengzhou-msft** [comment]
`az --version` checks for all updates of `azure-cli` related packages while `az upgrade` only updates when `azure-cli` has new versions. Normally, all related packages are released together. We had some issue with the pypi package of telemetry and had to release a new pypi version for it. The telemetry inside MSI/deb/RPM is not impacted and we did not release a new version for them. That's why you're seeing 1 updates available, but the telemetry package in MSI/deb/RPM actually has the latest functionality, no need to upgrade.

I have submitted #15152 to only show the instruction of using `az upgrade` when update for the `azure-cli` package is available to avoid confusion.
